cmd/mdatagen/metadata-schema.yaml

```diff
@@ -1,6 +1,17 @@
-# Required: type of the receiver.
+# Required: The type of the component.
 type:
```

Comment:
What is `type` and how does it differ from `class`?

Reply:
See here for context: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20264#discussion_r1149251957

Reply:
See this code as pointed from the link above: https://github.com/open-telemetry/opentelemetry-collector/blob/ff0dd4ce848dc6583609452f114f4b6eca423fb5/component/identifiable.go#L27